pkg/rpcservice/logs_tail.go

```diff
@@ -0,0 +1,199 @@
+package rpcservice
+
+import (
+	"context"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/stripe/stripe-cli/pkg/logtailing"
+	"github.com/stripe/stripe-cli/rpc"
+)
+
+// ITailer enables mocking a tailer object in tests
+type ITailer interface {
+	Run(context.Context) error
+}
+
+var createTailer = func(cfg *logtailing.Config) ITailer {
+	return logtailing.New(cfg)
+}
+
+// LogsTail returns a stream of API logs
+func (srv *RPCService) LogsTail(req *rpc.LogsTailRequest, stream rpc.StripeCLI_LogsTailServer) error {
+	deviceName, err := srv.cfg.UserCfg.Profile.GetDeviceName()
+	if err != nil {
+		return err
+	}
+
+	key, err := srv.cfg.UserCfg.Profile.GetAPIKey(false)
+	if err != nil {
+		return err
+	}
+
+	filters := getFiltersFromReq(req)
+
+	logtailingVisitor := createVisitor(&stream)
+
+	logtailingOutCh := make(chan logtailing.IElement)
+
+	logger := log.StandardLogger()
+
+	tailer := createTailer(&logtailing.Config{
+		DeviceName: deviceName,
+		Filters:    filters,
+		Key:        key,
+		Log:        logger,
+		OutCh:      logtailingOutCh,
+
+		// Hidden for debugging
+		APIBaseURL: "",
+		NoWSS:      false,
+	})
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go tailer.Run(ctx)
+
+	for {
+		select {
+		case e := <-logtailingOutCh:
+			err := e.Accept(logtailingVisitor)
+			if err != nil {
+				return err
+			}
+		case <-stream.Context().Done():
+			return stream.Context().Err()
+		}
+	}
+}
+
+func createVisitor(stream *rpc.StripeCLI_LogsTailServer) *logtailing.Visitor {
+	return &logtailing.Visitor{
+		VisitError: func(ee logtailing.ErrorElement) error {
+			return ee.Error
+		},
+		VisitLog: func(le logtailing.LogElement) error {
+			logError := rpc.LogsTailResponse_Log_Error{
```

Comment:
will `VisitLog` always be called with an error? I'm confused here.

Reply:
No, it won't always be called with an error. It's probably better if I change it to send `Error: nil` if there's no error.